pkg/kudoctl/cmd/init/webhook.go

```diff
@@ -0,0 +1,160 @@
+package init
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/util/kudo"
+	"k8s.io/api/admissionregistration/v1beta1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	clientv1beta1 "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	"sigs.k8s.io/yaml"
+)
+
+// Install uses Kubernetes client to install KUDO manager prereqs.
+func installWebhook(client kubernetes.Interface, dynamicClient dynamic.Interface, ns string) error {
+	if err := installUnstructured(dynamicClient, certificate(ns)); err != nil {
+		return err
+	}
+	if err := installAdmissionWebhook(client.AdmissionregistrationV1beta1(), admissionWebhook(ns)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func installUnstructured(dynamicClient dynamic.Interface, items []unstructured.Unstructured) error {
+	for _, item := range items {
+		obj := item
+		gvk := item.GroupVersionKind()
+		_, err := dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: fmt.Sprintf("%ss", strings.ToLower(gvk.Kind)), // since we know what kinds are we dealing with here, this is OK
+		}).Namespace(obj.GetNamespace()).Create(&obj, metav1.CreateOptions{})
+		if kerrors.IsAlreadyExists(err) {
+			clog.V(4).Printf("resource %s already registered", obj.GetName())
+		} else if err != nil {
+			return fmt.Errorf("Error when creating resource %s/%s. %v", obj.GetName(), obj.GetNamespace(), err)
+		}
+	}
+	return nil
+}
+
+func installAdmissionWebhook(client clientv1beta1.ValidatingWebhookConfigurationsGetter, webhook v1beta1.ValidatingWebhookConfiguration) error {
+	_, err := client.ValidatingWebhookConfigurations().Create(&webhook)
+	if kerrors.IsAlreadyExists(err) {
```

Comment:
In this case, should we Get the webhook configuration and deep-compare if it matches the expected configuration? Thinking about changes in the webhook, for example when we want to validate `operatorVersions` in addition to `instances`

Reply:
this is along the lines we handle all other resources right now. If we want to change it, I would do it for all and in another PR... but in general I agree :)